compiler/src/iree/compiler/Codegen/LLVMGPU/TransformExtensions/LLVMGPUExtensions.cpp

```diff
@@ -64,6 +65,179 @@ void transform_dialect::MapNestedForallToGpuThreadsOp::build(
   result.addTypes({pdl::OperationType::get(ctx)});
 }
 
+/// Return a flatten thread id for the workgroup with given sizes.
+static Value getFlatId(Location loc, RewriterBase &rewriter,
```

Comment:
Note, this seems : https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/Dialect/Utils/IndexingUtils.h#L91
(we may want to update the upstream API a bit, builder as last argument is ugly, my bad ..)